api/src/main/java/org/openmrs/module/fhir2/providers/r3/LocationFhirResourceProvider.java

```diff
@@ -57,6 +61,38 @@ public Location getLocationById(@IdParam @NotNull IdType id) {
 		return Location30_40.convertLocation(location);
 	}
 	
+	
+	@Create
+	public MethodOutcome createLocation(@ResourceParam Location location) {
+		return FhirProviderUtils.buildCreate(locationService.create(Location30_40.convertLocation(location)));
+	} 
+
+	@Update
+	@SuppressWarnings("unused")
+	public MethodOutcome updateLocation(@IdParam IdType id, @ResourceParam Location location) {
+		if (id == null || id.getIdPart() == null) {
+			throw new InvalidRequestException("id must be specified to update");
+		}
+
+		location.setId(id.getIdPart());
+
+		return FhirProviderUtils.buildUpdate(locationService.update(id.getIdPart(),
+		    Location30_40.convertLocation(location)));
+	}
+
+	@Delete
+	@SuppressWarnings("unused")
+	public OperationOutcome deleteLocation(@IdParam @NotNull IdType id) {
+		org.hl7.fhir.r4.model.Location location = locationService.delete(id.getIdPart());
+		if (location == null) {
+			throw new ResourceNotFoundException("Could not find location to delete with id " + id.getIdPart());
+		}
+		OperationOutcome retVal = new OperationOutcome();
+		retVal.setId(id.getIdPart());
+		retVal.getText().setDivAsString("Deleted successfully");
+		return retVal;
```

Comment:
Could you update this to use the `FhirProviderUtils.buildDelete()` method for this?

Reply:
@ibacher  i have added `FhirProviderUtils.buildDelete()`

About the the create, delete, and update to the tests for the R3 version of the provider, the `Location` instance used in the R3 `LocationFhirResourceProviderTest` class is from `org.hl7.fhir.r4.model.Location` which brings about incompatibility mis-match in the arguments ie `The method updateLocation(org.hl7.fhir.dstu3.model.IdType, org.hl7.fhir.dstu3.model.Location) in the type LocationFhirResourceProvider is not applicable for the arguments (org.hl7.fhir.dstu3.model.IdType, org.hl7.fhir.r4.model.Location)`..
I have tried changing the Location instance in R3 `LocationFhirResourceProviderTest` to `org.hl7.fhir.dstu3.model.Location` in order to match the method arguments but it break the existing tests because they use methods defined in `org.hl7.fhir.r4.model.Location` which are not defined in `org.hl7.fhir.dstu3.model.Location`

Reply:
the reason as to why the tests work in R4 is because it uses `org.hl7.fhir.r4.model.Location` location instance

Reply:
@gitcliff So I would use something like this to test the create method for instance:

```java
@Test
public void createLocation_shouldCreateNewLocation() {
	when(locationService.create(any())).thenReturn(location);
	org.hl7.fhir.dstu3.model.Location r3Location = Location30_40.convertLocation(location);

	MethodOutcome result = resourceProvider.createLocation(r3Location);

	assertThat(result, notNullValue());
	assertThat(result.getCreated(), is(true));
	assertThat(result.getResource().getIdElement().getIdPart(), equalTo(r3Location.getId()));
}
```

Note that using this did require some surgery. In particular, I needed to change the types in `FhirProviderUtils` to accept `IAnyResource` instead of `DomainResource`, but this should be a fairly harmless change.

Reply:
hello @ibacher thanks for the response above...

i came up with a test case for the update method like this one 

```
@Test
	public void updateLocation_shouldUpdateLocation() {
		
		when(locationService.update(LOCATION_UUID, location)).thenReturn(location);
		
		org.hl7.fhir.dstu3.model.Location r3Location = new org.hl7.fhir.dstu3.model.Location();
		r3Location = Location30_40.convertLocation(location);
		
		MethodOutcome result = resourceProvider.updateLocation(new IdType().setValue(LOCATION_UUID), r3Location);
		assertThat(result, CoreMatchers.notNullValue());
		assertThat(result.getResource(), equalTo(r3Location));
	}
	
```
however it fails with this [error ](https://pastebin.com/Jg8fdK8b), i did use `IAnyResource` instead of `DomainResource` in `FhirProviderUtils` not sure what i could be missing here